logictest: failure in distsql_stats for virtual columns

pkg/sql/logictest/tests/5node/5node_test.TestLogic_distsql_stats [failed](https://tanzanite.cluster.engflow.com/invocations/default/b8827329-ddca-46a9-9b31-a41b550a4bfb?testReportRun=7&testReportShard=11&testReportAttempt=1#targets-Ly9wa2cvc3FsL2xvZ2ljdGVzdC90ZXN0cy81bm9kZTo1bm9kZV90ZXN0) on release-24.1 @ [5d952f80b3e1efe2e9aaed73f1fd68433880fcb7](https://github.com/cockroachdb/cockroach/commits/5d952f80b3e1efe2e9aaed73f1fd68433880fcb7):


```
    logic.go:3021: let $only_null_stat = [{"avg_size": 0, "columns": ["a"], "created_at": "2024-03-31 06:16:43.778278", "distinct_count": 1, "histo_col_type": "INT8", "histo_version": 3, "name": "only_null_stat", "null_count": 3, "row_count": 3}]
    logic.go:3021: let $hist_id_1 = 956155180531318789
    logic.go:3021: let $hist_d = 956155181367066629
    logic.go:3021: let $hist_e = 956155181506822149
    logic.go:3021: let $hist_f = 956155181652082693
[06:16:44] --- progress: /var/lib/engflow/worker/work/3/exec/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/5node/5node_test_/5node_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/distsql_stats: 340 statements
    logic.go:3021: let $hist_crdb_internal_idx_expr = 956155182815477765
    logic.go:3021: let $hist_d = 956155182964211717
    logic.go:3021: let $hist_e = 956155183102590981
    logic.go:3021: let $hist_crdb_internal_idx_expr = 956155183254274053
[06:17:01] --- progress: /var/lib/engflow/worker/work/3/exec/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/5node/5node_test_/5node_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/distsql_stats: 357 statements
    logic.go:2975: 
         
        /var/lib/engflow/worker/work/3/exec/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/5node/5node_test_/5node_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/distsql_stats:2808: EXPLAIN SELECT * FROM mno WHERE n = 1 AND o = 9
        expected:
            distribution: full
            vectorized: true
            ·
            • render
            │
            └── • filter
                │ estimated row count: 1
                │ filter: n = 1
                │
                └── • scan
                      estimated row count: 18 (1.8% of the table; stats collected <hidden> ago)
                      table: mno@mno_o_idx
                      spans: [/9 - /9]
            
        but found (query options: "") :
            distribution: full
            vectorized: true
            ·
            • render
            │
            └── • filter
                │ filter: sqrt(m::FLOAT8)::INT8 = 9
                │
                └── • scan
                      missing stats
                      table: mno@mno_n_idx
                      spans: [/1 - /1]
            
[06:17:01] --- done: /var/lib/engflow/worker/work/3/exec/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/5node/5node_test_/5node_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/distsql_stats with config 5node: 366 tests, 1 failures
    logic.go:4145: 
        /var/lib/engflow/worker/work/3/exec/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/5node/5node_test_/5node_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/distsql_stats:2825: error while processing
    logic.go:4145: /var/lib/engflow/worker/work/3/exec/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/5node/5node_test_/5node_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/distsql_stats:2825: too many errors encountered, skipping the rest of the input
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestLogic_distsql_stats2861301879
--- FAIL: TestLogic_distsql_stats (30.14s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=7</code>
 - <code>shard=11</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-queries
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLogic_distsql_stats.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>


Jira issue: CRDB-37243

--- a/pkg/sql/logictest/testdata/logic_test/distsql_stats
+++ b/pkg/sql/logictest/testdata/logic_test/distsql_stats
@@ -2807,7 +2807,7 @@ NULL  {m}  1000  1000  true
 NULL  {n}  1000  50    true
 NULL  {o}  1000  33    true
 
-query T
+query T retry
 EXPLAIN SELECT * FROM mno WHERE n = 1 AND o = 9
 ----
 distribution: full

